### ISSUE TITLE
perspective correction: control lines need more click for them to be displayed

### ISSUE THREAD

**Issue by TurboGit** [darktable-org/darktable#13409]
**Describe the bug/issue**

When you click on the automatic detection of lines on the perspective correction module nothing happens. Only after 3 clicks or more the lines are displayed.

**To Reproduce**

1. Go to 'darkroom'
2. Enable 'perspective correction'
3. Click on '|||' (find control lines)
5. See error, no control lines displayed
6. click again 2 or more times and the lines are displayed

**Expected behavior**

The controls lines should be displayed first time.

**Which commit introduced the error**

```
062f087b5258db3c70792356dafd173adc923262 is the first bad commit
commit 062f087b5258db3c70792356dafd173adc923262
Author: Diederik ter Rahe <dterrahe@yahoo.com>
Date:   Wed Jan 4 09:17:26 2023 -0500

    don't recalculate on crop change when focused

 src/develop/develop.c | 3 ++-
 src/develop/develop.h | 1 +
 src/iop/ashift.c      | 5 +++++
 src/iop/crop.c        | 2 ++
 4 files changed, 10 insertions(+), 1 deletion(-)
```

**Platform**
_Please fill as much information as possible in the list given below. Please state "unknown" where you do not know the answer and remove any sections that are not applicable _

* darktable version : master
* OS                : GNU/Linux
* Linux - Distro    : GNU/Debian - sid

**Comment by TurboGit** [comment]
@dterrahe : Bisected to one of you commit :) Can you have a look? TIA.

**Comment by TurboGit** [comment]
Closing as duplicate of #13404.

**Comment by TurboGit** [comment]
If fact this is close to #13404, and point to the same bad commit.

**Comment by dterrahe** [comment]
> bad commit.

I know this is just a term from git, but I would argue _that_ commit is not "bad", it just stopped papering over bad code elsewhere...

**Comment by TurboGit** [comment]
Yes right, that's Git term don't want to offense you in any way.

**Comment by dterrahe** [comment]
> Yes right, that's Git term don't want to offense you in any way.

_Or_ offend that particular commit! Some commits can be very sensitive about that kind of thing. And this was a very _good_ commit (just not a _perfect_ one...)

More seriously, when ripping off band-aids that cover up all kinds of problems in code, git bisect tends to be less useful, because the real problems were introduced much longer ago.
